nautobot/core/utils/config.py

```diff
@@ -1,19 +1,39 @@
 """Helper code for loading values that may be defined in settings.py/nautobot_config.py *or* in django-constance."""
 
 import contextlib
+import logging
 
 from constance import config
 from django.conf import settings
 from django.core.exceptions import ObjectDoesNotExist
 from django.db import OperationalError, ProgrammingError
 
+logger = logging.getLogger(__name__)
 
-def get_settings_or_config(variable_name):
-    """Get a value from Django settings (if specified there) or Constance configuration (otherwise)."""
+
+def get_settings_or_config(variable_name, fallback=None):
+    """
+    Get a value from Django settings (if specified there) or Constance configuration (otherwise).
+
+    The fallback value is returned *only* if the requested variable cannot be found at all - this is an error case,
+    and will generate warning logs.
+    """
     # Explicitly set in settings.py or nautobot_config.py takes precedence, for now
     if hasattr(settings, variable_name):
         return getattr(settings, variable_name)
     # django-constance 4.x removed some built-in error handling here, so we have to do it ourselves now
     with contextlib.suppress(ObjectDoesNotExist, OperationalError, ProgrammingError):
         return getattr(config, variable_name)
```

Comment:
For my own edification, what is the difference between `config` here and `settings.CONSTANCE_CONFIG`?

Reply:
`config` is the Constance API object that provides access to the database table containing the current configured values for all Constance settings.

`settings.CONSTANCE_CONFIG` is what defines the records that *should* be in that table and their default/initial values, description, etc.